### ISSUE TITLE
RelayCommand Source Generator - Misleading error when using nullable reference type command parameter

### ISSUE THREAD

**Issue by mikechristiansenvae** [CommunityToolkit/dotnet#350]
### Describe the bug

When the following conditions are met, a misleading error message is produced.

- The `RelayCommandAttribute` source generator attribute is used
- The `CanExecute` property is set
- The ☝️ `CanExecute` method has a nullable reference type parameter

The error message:

>   [MVVMTK0011] The CanExecute name must refer to a compatible member, but no valid members were found for "CanNavigateForward" 

**Details:**

The check that is failing is [located here](https://github.com/CommunityToolkit/dotnet/blob/8528092ed255478db6d794a92829311ed43a52d3/CommunityToolkit.Mvvm.SourceGenerators/Input/RelayCommandGenerator.Execute.cs#L738), resulting in the diagnostic message being [reported here](https://github.com/CommunityToolkit/dotnet/blob/8528092ed255478db6d794a92829311ed43a52d3/CommunityToolkit.Mvvm.SourceGenerators/Input/RelayCommandGenerator.Execute.cs#L746).

In `CommunityToolkit.Mvvm.SourceGenerators.RelayCommandGenerator.Execute.TryGetCanExecuteExpressionFromSymbol`, there are the following failure cases:

- If it's a method:
   - Method return type is not `bool`
   - Parameter count is not 0 or 1
   - If there is a parameter, its type's fully qualified name is not equal to the fully qualified name of the type of the command's type argument.  (FYI, 👈 [I suspect that this check is what's causing this bug](https://github.com/CommunityToolkit/dotnet/blob/8528092ed255478db6d794a92829311ed43a52d3/CommunityToolkit.Mvvm.SourceGenerators/Input/RelayCommandGenerator.Execute.cs#L795) - something is not accounting for nullable reference types)
- If it's a property:
   - It must be a `bool`

_However_ - if the `CanExecute` doesn't meet any of those conditions - _the user is never informed which one._

**Suggestion:**

The diagnostic messages should be expanded to indicate _why_ the `CanExecute` method was not supported.  For example, if the user provides a method that returns an `int`, the message should indicate that.

### Steps to reproduce

```text
The below code produces the error:


[RelayCommand(CanExecute = nameof(CanNavigateForward))]
protected virtual void NavigateForward(string? parameter)
{
    ;
}

protected virtual bool CanNavigateForward(string? parameter)
{
    return true;
}


- Using a nullable *value* type (`int`, `bool`, etc.) as the parameter does *not* produce the error.  
- Making the parameter non-nullable (i.e., removing the `?`) does *not* produce the error.
```

### Expected behavior

If I, as the developer, have defined the method parameter as a nullable reference type in the `Execute` method, then:

- the `RelayCommand` implementation should allow `null` command parameters to be sent to the `Execute` method.
- the source generator should allow the `CanExecute` method to accept nullable parameters.

### IDE and version

Rider

### IDE version

Rider 2022.1.2 #RD-221.5787.36

### Nuget packages

- [X] CommunityToolkit.Common
- [X] CommunityToolkit.Diagnostics
- [ ] CommunityToolkit.HighPerformance
- [X] CommunityToolkit.Mvvm (aka MVVM Toolkit)

### Nuget package version(s)

8.0.0-preview4

### Help us help you

No, just wanted to report this

**Comment by Sergio0694** [comment]
I suspect this might just be a duplicate of #294, which was fixed by #302.
Could you try with the latest version from the preview feed with builds from main? 🙂

**Comment by Sergio0694** [comment]
Closing as duplicate since we never heard back from the author. Feel free to reply if anything changes 🙂
